compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -950,4 +943,131 @@ private void addProvidedDependencyWarning(List<Diagnostic> emitResultDiagnostics
                     this.packageContext().descriptor().name().toString()));
         }
     }
+
+    private PlatformLibrary codeGeneratedResourcesLibrary(PackageId packageId, PlatformLibraryScope scope) {
+        Package pkg = packageCache.getPackageOrThrow(packageId);
+        CompilationCache compilationCache = pkg.project().projectEnvironmentContext().getService(
+                CompilationCache.class);
+        return compilationCache.getPlatformSpecificLibrary(this,
+                                RESOURCE_DIR_NAME)
+                .map(path -> new JarLibrary(path, scope))
+                .orElse(null);
+    }
+
+    private Map<String, byte[]> getResources(PackageContext packageContext) {
+        Map<String, byte[]> resourceMap = new HashMap<>();
+        for (DocumentId documentId : packageContext.resourceIds()) {
+            String resourceName = RESOURCE_DIR_NAME + "/"
+                    + packageContext.resourceContext(documentId).name();
+            resourceMap.put(resourceName, packageContext.resourceContext(documentId).content());
+        }
+        return resourceMap;
+    }
+
+    private Map<String, byte[]> getAllResources(PackageContext packageContext) {
```

Comment:
I think having names like `getPackageResources` and `getPackageAndTestResources` would be better because `getResources` and `getAllResources` are confusing. Without examining the method implementations, it's difficult to understand the differences between them.